### DISCUSSION TITLE
GUI for Touch screen

### DISCUSSION THREAD

**Discussion by muhammadelmogy** [vlachoudis/bCNC#1717] [Ideas]
I am trying to run the software on Raspberry pi with 7" display and the icons and buttons are small for touch and the finger can easily mis-touch unintended buttons
I was wondering if there is any chance to redesign the GUI to be more suitable for small touch screen or even big one
I have opened the code in Visual studio(I am a .net developer) and was able to change some basic  properties for the buttons like for example the padding 
`b.grid(row=row, column=col, padx=0, pady=0, sticky=EW)`
 to 
`b.grid(row=row, column=col, padx=20, pady=20, sticky=EW)`

This gave the button a big space around it and made it more appropriate for touch but I do not think this is the right way to change the GUI

I don't have any experience developing with Python or Desktop Applications with Python and I want to do this and if possible make it available as a a second GUI that user can choose at installation

Does the GUI use a separate files for the layout(could not find any) or should I directly modify the files that ends with Page.py(or/and others)?

I hope someone can give a general directions on how to do this properly and in way that would make it easy to test and does not affect the functionality of the software since I do not want to mess with internal code structure and stay away from it.

**Comment by metropolicon** [comment]
> I hope someone can give a general directions on how to do this properly and in way that would make it easy to test and does not affect the functionality of the software since I do not want to mess with internal code structure and stay away from it.

Hi...
May be you'll be interested by my bCNC addon (https://github.com/vlachoudis/bCNC/discussions/1769)
Work on the same configuration like you ;)
